internals/cli/cli.go

```diff
@@ -63,52 +63,34 @@ var optionsData options
 // ErrExtraArgs is returned  if extra arguments to a command are found
 var ErrExtraArgs = fmt.Errorf("too many arguments for command")
 
-// cmdInfo holds information needed to call parser.AddCommand(...).
-type cmdInfo struct {
-	name, shortHelp, longHelp string
-	builder                   func() flags.Commander
-	hidden                    bool
-	optDescs                  map[string]string
-	argDescs                  []argDesc
-	alias                     string
-	extra                     func(*flags.Command)
+// CmdInfo holds information needed to call parser.AddCommand(...).
+type CmdInfo struct {
+	Name, ShortHelp, LongHelp string
+	Builder                   func() flags.Commander
+	Hidden                    bool
+	OptDescs                  map[string]string
+	ArgDescs                  []argDesc
+	Alias                     string
+	Extra                     func(*flags.Command)
 }
 
 // commands holds information about all non-debug commands.
-var commands []*cmdInfo
+var commands []*CmdInfo
 
 // debugCommands holds information about all debug commands.
-var debugCommands []*cmdInfo
+var debugCommands []*CmdInfo
 
 // AddCommand replaces parser.addCommand() in a way that is compatible with
 // re-constructing a pristine parser.
-func AddCommand(name, shortHelp, longHelp string, builder func() flags.Commander, optDescs map[string]string, argDescs []argDesc) *cmdInfo {
-	info := &cmdInfo{
-		name:      name,
-		shortHelp: shortHelp,
-		longHelp:  longHelp,
-		builder:   builder,
-		optDescs:  optDescs,
-		argDescs:  argDescs,
-	}
+func AddCommand(info *CmdInfo) {
```

Comment:
This is a much better API, but what if the caller changes `info` later on?

Reply:
Agreed, we should instead pass in `info` by value.